lib/player.js

```diff
@@ -2002,7 +2002,28 @@ shaka.Player = class extends shaka.util.FakeEventTarget {
       // TODO: File a bug or send a PR to the compiler externs to fix this.
       const textTracks = /** @type {EventTarget} */(this.video_.textTracks);
       this.eventManager_.listen(
-          textTracks, 'addtrack', () => this.onTracksChanged_());
+          textTracks, 'addtrack', (e) => {
+            this.onTracksChanged_();
+            const track = e.track;
+            if (track.kind != 'metadata') {
+              return;
+            }
+            track.mode = 'hidden';
+            this.eventManager_.listen(track, 'cuechange', () => {
+              for (const cue of track.activeCues) {
+                if (this.adManager_) {
+                  this.adManager_.onCueMetadataChange(cue.value);
+                }
+              }
+            });
+            new shaka.util.Timer(() => {
+              const textTracks = Array.from(this.video_.textTracks)
+                  .filter((track) => { return track.kind == 'metadata'; });
+              for (const textTrack of textTracks) {
+                textTrack.mode = 'hidden';
+              }
+            }).tickNow().tickAfter(/* seconds= */ 0.5);
```

Comment:
This is why I asked about the timer. Could you explain this part to me please.

Reply:
Ah sorry! This timer is needed because Safari is not always setting the mode to hidden, so it's necessary verify in several times that this is done.